ipie/addons/free_projection/propagation/CCSD.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2022 The ipie Developers. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors: Fionn Malone <fmalone@google.com>
+#          Joonho Lee
+#
+
+from dataclasses import dataclass
+
+from ipie.utils.backend import arraylib as xp
+
+
+@dataclass
```

Comment:
It's a little weird to me that a dataclass  has an `__init__` method typically you would just have a factory method which does any instantiate it.

Reply:
okay this is no longer a dataclass